src/rpc/mining.cpp

```diff
@@ -166,7 +166,7 @@ static UniValue generateBlocks(ChainstateManager& chainman, CEvoDB& evodb, CChai
     UniValue blockHashes(UniValue::VARR);
     while (nHeight < nHeightEnd && !ShutdownRequested())
     {
-        std::unique_ptr<CBlockTemplate> pblocktemplate(BlockAssembler(chainman.ActiveChainstate(), evodb, chain_helper, mnhfman, llmq_ctx, mempool, Params()).CreateNewBlock(coinbase_script));
+        std::unique_ptr<CBlockTemplate> pblocktemplate(BlockAssembler(chainman.ActiveChainstate(), cpoolman, evodb, chain_helper, mnhfman, llmq_ctx, mempool, Params()).CreateNewBlock(coinbase_script));
```

Comment:
for `m_cpoolman` consider to make a helper method `CChainstateHelper::GetCreditPoolManger()` instead passing `cpoolman` directly to BlockProcessor.

It would be more error prune and ensure that BlockAssembler will use exactly same `cpoolman` as CChainstate + simplify BlockAssembler constructor

Reply:
I might have been okay with this if:
  * `chain_helper` was a collection of functions (as an earlier revision was, using inheritance)
  * We're proxying only individual functions (like `GetMNHFSignalsStage`) and not the manager itself

But since it's been made an aggregate/context (collection of helpers), I'm not quite confident about it being the best approach.

Since we're embracing passing `NodeContext` as a whole (47de7deeb17af8c541284721e387231342d45acb), we're doing the ref differentiation in construction and reducing the opportunity to pass the wrong item as an argument. If `NodeContext` itself is misconfigured, there are bigger problems.